cmd/gossamer/gossamer_test.go

```diff
@@ -0,0 +1,154 @@
+package main_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/build"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/p2p"
+	"github.com/ChainSafe/gossamer/polkadb"
+	"github.com/inconshreveable/log15"
+	"github.com/rendon/testcli"
+)
+
+var binaryname = "gossamer-test"
+
+const configTest = "config-test.toml"
+const timeFormat = "2006-01-02T15:04:05-0700"
+
+func setup() {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+	}
+	r := exec.Command("go", "build", "-o", gopath+"/bin/gossamer-test")
+	err := r.Run()
+	if err != nil {
+		log15.Crit("could not execute binary", "executable", binaryname, "err", err)
+		os.Exit(1)
+	}
+	run := exec.Command(`gossamer-test`)
+	err = run.Run()
+	if err != nil {
+		log15.Crit("could not execute binary", "executable", binaryname, "err", err)
+		os.Exit(1)
+	}
+	fp, err := os.Create(configTest)
+	if err != nil {
+		log15.Crit("could not create test config", "config", configTest, "err", err)
+		os.Exit(1)
+	}
+
+	testConfig := fmt.Sprintf("%s%s%s%v%s%s%v%s%s%s",
```

Comment:
This is very unmaintainable as we add new flags. I think this should at least be moved into a struct that gets parsed into a string

Reply:
I think `expectedResponses` could also benefit from some of this.

Reply:
On lines 132 `testConfig` struct in `expectedResponses` suffice for maintainability? If so, I will proceed to implement the others accordingly, if not I may need an alternative suggestion

Reply:
bump @ansermino

Reply:
yes, exactly

Reply:
@ansermino cannot seem to utilize structs and then write to temp config file due to the strict yaml syntax, any other suggestions?

Reply:
Shoot me a message with what you've tried and I'll take a look

Reply:
lgtm and revisit with rest of the config?